### PR TITLE
Expose points layer antialiasing control publicly

### PR DESCRIPTION
**Pull request by alisterburt**
# Description
This PR exposes our previously private `Points._antialias` attribute as `Points.antialiasing` as a fix for #4683 

@brisvag could you check the default behaviour on layer construction and make sure you're happy with it? You're more familiar with this than I am

@sofroniewn in the issue @brisvag mentioned you might not like this so cc'ing you here

@dagewa hope this gets you where you wanted to be!

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: napari/layers/points/points.py [hunk 1]
```diff
@@ -447,7 +450,7 @@ def __init__(
 
         self.experimental_canvas_size_limits = experimental_canvas_size_limits
         self.shading = shading
-        self._antialias = True
+        self.antialiasing = 1 if antialiasing is None else antialiasing
```
Comment (brisvag) [inline napari/layers/points/points.py]: Why this if clause, and why 1 if None? I would have just put 1 as default and have no clause here, but maybe I'm missing something? Or at least 0 if None...

Comment (alisterburt) [reply napari/layers/points/points.py]: I was following the pattern of `None` being the default kwarg for most things in the constructor

Comment (brisvag) [reply napari/layers/points/points.py]: Not sure why that's the case, actually; I would say it makes sense only in a few cases (such as with complicated defaults like affines, mutable things like metadata, or autogenerated defaults like `name`), but I don't see why it should be the case here (or in stuff like `n_dimensional`, for example).

**Comment by alisterburt** [comment]
okay @brisvag have changed the kwarg to default to 1 - I think the pattern of `None` is there so that different behaviours in different `_ndim`s can be handled gracefully but that's not what's happening here. As before, I'm not sure the default of 1 is appropriate in all cases, it looks good on my machine -> if we instantiate a points layer with your 'spherical' shading then perhaps we would prefer to turn antialiasing off by default?

edit: test failure is unrelated https://github.com/napari/napari/runs/7062118061?check_suite_focus=true#step:8:218

**Comment by brisvag** [comment]
> if we instantiate a points layer with your 'spherical' shading then perhaps we would prefer to turn antialiasing off by default?

This was also proposed when I introduced the shader. I'm not in love with defaults magically changing like, this, but I think I might be in the minority here :) I think @sofroniewn and probably also @jni where in favor last time?

**Review by Carreau — APPROVED** [review]
No objections from me, restarted the failed test, that appear to be spurious failure.

File: napari/layers/points/points.py [hunk 2]
```diff
@@ -762,16 +765,23 @@ def current_size(self, size: Union[None, float]) -> None:
...
+        """
         if value < 0:
-            value = 0
-        self.__antialias = float(value)
-        self.events._antialias()
+            warnings.warn(
+                message='antialiasing amount must be positive, value will be clipped.',
+                category=RuntimeWarning,
+            )
+        self._antialiasing = float(np.clip(value, a_min=0, a_max=None))
```
Comment (Czaki) [inline napari/layers/points/points.py]:
```suggestion
        self._antialiasing = max(0, value)
```

File: napari/layers/points/points.py [hunk 3]
```diff
@@ -762,16 +765,23 @@ def current_size(self, size: Union[None, float]) -> None:
...
+        self.events.antialiasing()
```
Comment (Czaki) [inline napari/layers/points/points.py]:
```suggestion
        self.events.antialiasing(value=self._antialiasing)
```

**Review by Czaki — APPROVED** [review]
two minor suggestions

File: napari/layers/points/points.py [hunk 4]
```diff
@@ -762,16 +765,23 @@ def current_size(self, size: Union[None, float]) -> None:
...
+        """Set the amount of antialiasing in pixels.
+
+        Values can only be positive.
+        """
         if value < 0:
-            value = 0
-        self.__antialias = float(value)
-        self.events._antialias()
+            warnings.warn(
+                message='antialiasing amount must be positive, value will be clipped.',
```
Comment (kevinyamauchi) [inline napari/layers/points/points.py]:
```suggestion
                message='antialiasing value must be positive, value will be set to 0.',
```

**Review by kevinyamauchi — APPROVED** [review]
This looks good to me! Minor suggestion below.

File: napari/layers/points/points.py [hunk 5]
```diff
@@ -142,6 +142,8 @@ class Points(Layer):
           No shading is added to the points.
         * 'spherical'
           Shading and depth buffer are changed to give a 3D spherical look to the points
+    antialiasing: float
+        Amount of antialiasing in pixels.
```
Comment (brisvag) [inline napari/layers/points/points.py]: for consistency with other docstrings, I would say "canvas pixels".

File: napari/layers/points/points.py [hunk 6]
```diff
@@ -241,6 +243,8 @@ class Points(Layer):
         COLORMAP allows color to be set via a color map over an attribute
     shading : Shading
         Shading mode.
+    antialiasing: float
+        Amount of antialiasing in pixels.
```
Comment (brisvag) [inline napari/layers/points/points.py]: same as above.

File: napari/layers/points/points.py [hunk 7]
```diff
@@ -762,16 +765,23 @@ def current_size(self, size: Union[None, float]) -> None:
             self.events.size()
 
     @property
-    def _antialias(self):
-        """float: amount in pixels of antialiasing"""
-        return self.__antialias
+    def antialiasing(self) -> float:
+        """Amount of antialiasing in pixels."""
+        return self._antialiasing
```
Comment (brisvag) [inline napari/layers/points/points.py]: same.

File: napari/layers/points/points.py [hunk 8]
```diff
@@ -762,16 +765,23 @@ def current_size(self, size: Union[None, float]) -> None:
...
 
-    @_antialias.setter
-    def _antialias(self, value) -> Union[int, float]:
+    @antialiasing.setter
+    def antialiasing(self, value: float):
+        """Set the amount of antialiasing in pixels.
```
Comment (brisvag) [inline napari/layers/points/points.py]: aaaand here too :)

**Review by brisvag — APPROVED** [review]

**Review by sofroniewn — APPROVED** [review]
LGTM! Feel free to fix merge conflict and then merge when tests passing @alisterburt

**Pull request merged**